pallets/creator-staking/src/lib.rs

```diff
@@ -0,0 +1,748 @@
+
+            // TODO: change event
+            Self::deposit_event(Event::<T>::RewardsClaimed {
+                who: staker,
+                amount: staker_reward,
+            });
+
+            /*Ok(Some(if should_restake_reward {
+                T::WeightInfo::claim_staker_with_restake()
+            } else {
+                T::WeightInfo::claim_staker_without_restake()
+            })
+                .into())*/
+            Ok(().into())
+        }
+
+        #[pallet::call_index(8)]
+        #[pallet::weight(Weight::from_ref_time(10_000))]
+        pub fn claim_creator_reward(origin: OriginFor<T>, space_id: SpaceId, era: EraIndex) -> DispatchResultWithPostInfo {
+            Self::ensure_pallet_enabled()?;
+            let _ = ensure_signed(origin)?;
+
+            let creator_info =
+                RegisteredCreators::<T>::get(space_id).ok_or(Error::<T>::NotRegisteredCreator)?;
+
+            let mut creator_stake_info =
+                Self::creator_stake_info(space_id, era).unwrap_or_default();
+
+            let creator_reward = Self::calculate_creator_reward(&creator_stake_info, &creator_info, era)?;
+
+            // Withdraw reward funds from the creators staking
+            let reward_imbalance = T::Currency::withdraw(
+                &Self::account_id(),
+                creator_reward,
+                WithdrawReasons::TRANSFER,
+                ExistenceRequirement::AllowDeath,
+            )?;
+
+            T::Currency::resolve_creating(&creator_info.stakeholder, reward_imbalance);
+
+            // TODO: change event
+            Self::deposit_event(Event::<T>::RewardsClaimed {
+                who: creator_info.stakeholder,
+                amount: creator_reward,
+            });
+
+            // updated counter for total rewards paid to the creator
+            creator_stake_info.creator_reward_claimed = true;
+            CreatorEraStake::<T>::insert(space_id, era, creator_stake_info);
+
+            Ok(().into())
+        }
+
+        #[pallet::call_index(9)]
+        #[pallet::weight(Weight::from_ref_time(10_000))]
+        pub fn set_maintenance_mode(
+            origin: OriginFor<T>,
+            enable_maintenance: bool,
+        ) -> DispatchResultWithPostInfo {
+            ensure_root(origin)?;
+            let is_disabled = PalletDisabled::<T>::get();
+
+            ensure!(
+                is_disabled ^ enable_maintenance,
+                Error::<T>::MaintenanceModeNotChanged
+            );
+            PalletDisabled::<T>::put(enable_maintenance);
+
+            Self::deposit_event(Event::<T>::MaintenanceModeSet { status: enable_maintenance });
+            Ok(().into())
+        }
+
+        #[pallet::call_index(10)]
+        #[pallet::weight(Weight::from_ref_time(10_000))]
+        pub fn force_new_era(origin: OriginFor<T>) -> DispatchResult {
+            Self::ensure_pallet_enabled()?;
+            ensure_root(origin)?;
+            ForceEra::<T>::put(Forcing::ForceNew);
+            Ok(())
```

Comment:
Need to add an event here.

Reply:
No need actually, because we already have `NewCreatorStakingEra`, which will show us that the era has started. Furthermore, this `ForceEra` thing is specifically plan-B option and will probably not be used manually, only as a part of some migration.